### PR TITLE
Update AdderToolbar components and styling

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR updates the shared components used by `AdderToolbar`, and updates its Tailwind styling to eliminate the need for external CSS. I'm focusing on this component because I expect we'll make some adjustments in this area soon for accessibility reasons.

This component has some of the most fiddly styling of any of our components. It was also one of the first I converted to Tailwind. Much has been learned since February, but it doesn't eliminate the reality that the hover behavior for the `AdderToolbar` is rather nuanced.

These changes:

* Bump `frontend-shared` to 5.4.1 to get the correct `AnnotateIcon`
* Use updated shared components and icons
* Refactor styles out of external SASS
    * Add a simple Tailwind utility style to apply `all: initial`, which is something we do for `AdderToolbar` (it's in a shadow DOM) and isn't available in stock Tailwind
    * Add a more complex Tailwind variant to address the very specific hover-styling needs of this component.
* Comment _the living daylights_ out of `AdderToolbar`'s styling

### CODE REVIEW

File: src/annotator/components/AdderToolbar.js [hunk 1]
```diff
@@ -34,55 +55,60 @@ function NumberIcon({ badgeCount }) {
...
-      name="pointer"
-      classes={classnames(
-        // Position the arrow in the horizontal center at the bottom of the
-        // container (toolbar). Note: the arrow is pointing up at this point.
-        'absolute left-1/2 -translate-x-1/2',
-        // Override `1em` width/height rules in `Icon` to size the arrow as
-        // its SVG dimensions dictate
-        'h-auto w-auto z-2',
-        'text-grey-3 fill-white',
+    <div
```
Comment (lyzadanger) [inline src/annotator/components/AdderToolbar.js:58]: Here's an example of me adding an additional parent component to handle the positioning, and not applying that styling directly to the icon component. This is less of a real concern with icons, which have no classes applied to them other than what you set on `className`, but it's more hygienic IMHO.

File: src/annotator/components/AdderToolbar.js [hunk 2]
```diff
@@ -34,55 +55,60 @@ function NumberIcon({ badgeCount }) {
...
+        // When the parent .group element is hovered (but this element itself is
+        // not), dim this button's text. This has the effect of dimming inactive
+        // buttons.
+        'group-hover:text-grey-5',
+        // When the parent .group element is hovered AND this element is
+        // hovered, this is the "active" button. Intensify the text color, which
+        // will also darken any descendant Icon
+        'hover:group-hover:text-grey-9',
+        // Set the .hover-group class for use by `NumberIcon`
+        'hover-group'
```
Comment (lyzadanger) [inline src/annotator/components/AdderToolbar.js:103]: This use case is a good example of both the pros and cons of `ButtonBase`. On the "pros" side, we automatically get consistent transitions and focus rings and whatnot, so those don't have to be managed manually, nor risk being inconsistent. On the other hand, I just kind of have to "know" that `ButtonBase` already sets a `flex` layout, so I just need to set `flex-col` to go columnar. But I can feel assured that padding and colors won't conflict, as the coalescing convention for base classes is never to style dimensions or colors...

File: src/annotator/components/AdderToolbar.js [hunk 3]
```diff
@@ -34,55 +55,60 @@ function NumberIcon({ badgeCount }) {
...
       )}
       onClick={onClick}
       title={title}
     >
-      {icon && <Icon classes="text-annotator-lg" name={icon} title={title} />}
+      {Icon && <Icon className="text-annotator-lg" title={title} />}
```
Comment (lyzadanger) [inline src/annotator/components/AdderToolbar.js:108]: At least `className` gives a hint that your classes are the only classes...

File: src/annotator/components/AdderToolbar.js [hunk 4]
```diff
@@ -148,20 +207,21 @@ export default function AdderToolbar({
...
         }
       )}
+      data-component="AdderToolbar"
       dir="ltr"
       style={{
         visibility: isVisible ? 'visible' : 'hidden',
       }}
     >
-      <div className="flex dim-items-on-hover">
+      <div className="flex group">
```
Comment (lyzadanger) [inline src/annotator/components/AdderToolbar.js:216]: I wish Tailwind's `group` class had a better name. It's so...vague.

File: tailwind.config.mjs [hunk 5]
```diff
@@ -199,8 +199,12 @@ export default {
       // class to the sidebar frame when it's collapsed. This modifier allows
       // sub-components to select for that state.
       addVariant('sidebar-collapsed', '.sidebar-collapsed &');
+      // Custom variant for a specific use case in which an element needs to
+      // style itself based on the hover states of TWO parents (.group and
+      // .hover-group). See `src/annotator/components/AdderToolbar`
+      addVariant('hover-group-hover', '.group:hover .hover-group:hover &');
```
Comment (lyzadanger) [inline tailwind.config.mjs:205]: I'm not gonna mince words: this is a tad fugly. I hope the comments everywhere help. I've just never quite run into a situation like this before, where the hover state of two ancestors has to be assessed.

**Review by robertknight — COMMENTED** [review]
Regarding `NumberIcon` colors, it looks to me like this particular use case can be solved without needing to deal with nested groups using `currentcolor`. The need for styling which depends on the states of multiple ancestors doesn't seem that unusual to me (eg. see Tailwind feature request I linked in the comments), and I think it is something we should be able to accomplish without needing a ton of documentation each time. A high-level description of what the user should see is always useful. The solutions I can think of / find online boil down to one of:

- Adding some support for labeled groups, with varying degrees of generality, which is basically what you did here. If we're doing this, I'd suggest giving the groups names that reflect their semantic role or level in the hierarchy.
- Setting CSS variables at the top of the group, which children can then consume. This allows the ancestors to set the context ("the current foreground color is $X") while individual components encapsulate how it is applied ("set the `color` property to $X"). `currentcolor` is a special case of this.

Other than `NumberIcon`, I suggested we stick with with `icon` as the prop name to match the pattern library, but continue to accept components as values.

The rest of the changes look good to me.

**Comment by lyzadanger** [comment]
@robertknight `currentColor`: keys to the kingdom here! Didn't think of it. Should reduce all of this complexity to minimal. Thanks.

**Comment by lyzadanger** [comment]
OK, "reboot" here, much simpler, using `currentColor`, no need for any real cleverness. A single `group` will do the trick overall.

**Review by robertknight — APPROVED** [review]
Revised version LGTM

**Pull request merged**